src/shogun/lib/StoppableSGObject.cpp

```diff
@@ -40,6 +41,13 @@ rxcpp::subscription CStoppableSGObject::connect_to_signal_handler()
 	return get_global_signal()->get_observable()->subscribe(subscriber);
 }
 
+void CStoppableSGObject::serialize_machine(std::string filename)
```

Comment:
i'm not so sure if get it why there's a specific interface serializing a "machine". especially when `CStoppableSGObject` is actually not even a machine, but an SGObject

Reply:
yep, the name is misleading, just call it "serialize".

Reply:
@shubham808 it's not about the name. it's more about that `CStoppableSGObject` is a `CSGObject` which already have a serialization functionality in it `CSGObject::save_serializable` `CSGObject::load_serializable` which should be sufficient to save any SGObject-like

Reply:
IMO this can be a member method as it will depend on the state of the stoppable object, namely the file that might be used for dumping the temporary state

Reply:
why a file at all? why not a bytestream? i really dont get this concept atm

Reply:
because we dont have a bytestream right now. Once we have a generic interface for serializing into whatever, we will use it here

Reply:
CFile* ?

Reply:
yes it should be CFile. @shubham808 has not updated the PR yet after our discussion.